enterprise/coderd/groups.go

```diff
@@ -72,16 +74,44 @@ func (api *API) postGroupByOrganization(rw http.ResponseWriter, r *http.Request)
 
 func (api *API) patchGroup(rw http.ResponseWriter, r *http.Request) {
 	var (
-		ctx               = r.Context()
-		group             = httpmw.GroupParam(r)
-		auditor           = api.AGPL.Auditor.Load()
-		aReq, commitAudit = audit.InitRequest[database.Group](rw, &audit.RequestParams{
-			Audit:   *auditor,
-			Log:     api.Logger,
-			Request: r,
-			Action:  database.AuditActionWrite,
-		})
+		ctx     = r.Context()
+		group   = httpmw.GroupParam(r)
+		auditor = api.AGPL.Auditor.Load()
 	)
+
+	var req codersdk.PatchGroupRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	var (
+		groupResourceInfo    = make(map[string][]string)
+		hasGroupMemberChange = len(req.AddUsers) != 0 || len(req.RemoveUsers) != 0
+	)
+
+	if hasGroupMemberChange {
+		currentMembers, currentMembersErr := api.Database.GetGroupMembers(ctx, group.ID)
+		if currentMembersErr != nil {
+			httpapi.InternalServerError(rw, currentMembersErr)
+			return
+		}
+
+		groupResourceInfo = createMemberMap(req, groupResourceInfo, currentMembers)
+	}
+
+	wriBytes, marshalErr := json.Marshal(groupResourceInfo)
+	if marshalErr != nil {
+		httpapi.InternalServerError(rw, marshalErr)
+	}
+	aReq, commitAudit := audit.InitRequest[database.Group](rw, &audit.RequestParams{
+		Audit:                *auditor,
+		Log:                  api.Logger,
+		Request:              r,
+		Action:               database.AuditActionWrite,
+		HasGroupMemberChange: hasGroupMemberChange,
+		GroupMemberLists:     wriBytes,
+	})
+
```

Comment:
A pain point here is that if this request fails sometime before line 115, we won't get an audit log signifying the failure. I'm not sure how to get around this, given we want to pass some computational stuff in (`GroupMemberLists`).

Reply:
What if instead of using `database.Group` here we instead made `[]database.GroupMember` auditable? I think this might solve a lot of the jankiness of needing hardcoded stuff for group members.

Reply:
This might take some work adding support for slices in the diff code.

Reply:
@coadler I was trying to avoid having another auditable resource; however, I'm down to make the change because I agree - it would simplify a lot. In general, do you have any opinions about how many auditable resources we should have? I don't want to bloat the feature but maybe that's less of a concern than I think it is.